src/serial.rs

```diff
@@ -478,6 +534,26 @@ mod tests {
         }
     }
 
+    #[derive(Default)]
+    struct ExampleSerialMetrics {
+        read_count: AtomicU64,
+        out_byte_count: AtomicU64,
+    }
+
+    // We define this for Arc<SerialExampleMetrics> because we expect to share a reference to
+    // the same object with other threads.
+    // TODO: We can use `Metric` from `vmm-sys-util` after it gets published.
```

Comment:
nit: It is already published.

Reply:
My bad; updated the commit to use `vmm_sys_util::metric::Metric`.